process/block/shardblock.go

```diff
@@ -1710,12 +1735,11 @@ func (sp *shardProcessor) receivedMetaBlock(headerHandler data.HeaderHandler, me
 			}
 		}
 
+		hasProofForMetablock := false
 		// attesting something
 		if sp.hdrsForCurrBlock.missingHdrs == 0 {
-			sp.hdrsForCurrBlock.missingFinalityAttestingHdrs = sp.requestMissingFinalityAttestingHeaders(
-				core.MetachainShardId,
-				sp.metaBlockFinality,
-			)
+			hasProofForMetablock = sp.hasProofForMetablock(metaBlockHash, metaBlock)
```

Comment:
is might be cleaner to keep old code here and set `hasProofForMetablock` to true initially, and then set it based on `HasProof` if flag is enabled